skyportal/openapi.py

```diff
@@ -79,14 +79,17 @@ def spec_from_handlers(handlers):
                 continue
 
             path_template = endpoint
-            path_template = re.sub('\(.*?\)\??', '{}', path_template)
-            path_template = re.sub('(?=[^/]{1}){}', '/{}', path_template)
+            path_template = re.sub('\(.*?\)\??', '/{}', path_template)
+            path_template = re.sub('(/)+', '/', path_template)
             path_parameters = path_template.count('{}')
 
             spec = yaml_utils.load_yaml_from_docstring(method.__doc__)
             parameters = list(inspect.signature(method).parameters.keys())[1:]
             parameters = parameters + (path_parameters - len(parameters)) * ['',]
 
+            if parameters[-1:] == [''] and path_template.endswith('/{}'):
```

Comment:
Question: doesn't this just remove the trailing slash, same as below?

Reply:
That only handles the case for the "multiple" paths - when we omit these two lines, we still get paths like this in the docs: `/api/comment/`

Reply:
So, I guess the idea would be to do the trailing slash removal later (i.e., remove the .sub above, and place it after parameter substitution).

Reply:
We can do that, yes, but we'll have to do it twice -- there are two cases below. If we do it this way, we only do it once, and don't have any repeated code. Would you prefer I do it twice below, for each of the cases?

Reply:
(I removed the later sub, as it's actually only needed here)